virtool/analyses/migrate.py

```diff
@@ -11,3 +11,18 @@ async def migrate_analyses(app: virtool.types.App):
 
     """
     await virtool.db.utils.delete_unready(app["db"].analyses)
+    await change_to_subtractions_list(app["db"])
+
+
+async def change_to_subtractions_list(db):
+    async for document in db.analyses.find():
+        if "subtraction" in document:
+            await db.analyses.update_one({"_id": document["_id"]}, {
+                "$set": {
+                    "subtractions": [document["subtraction"]["id"]]
+                },
+                "$unset": {
+                    "subtraction": ""
+                }
+            })
```

Comment:
This will be very slow for large analyses collections. You can use a query instead of the conditional at line 19:
```python
async for document in db.analyses.find({"subtraction": {"$exists": True}}):
```

Using [bulk operations](https://motor.readthedocs.io/en/stable/examples/bulk.html) would also greatly speed this up.